### PR TITLE
2846: Restrict option set multilevel creation on modal.

### PR DESCRIPTION
**Pull request by lmatiolis**
Now we are disabling the option to be multilevel if it's creating an option set for a select multiple question.

### CODE REVIEW

File: app/assets/javascripts/views/question_form.js [hunk 1]
```diff
@@ -91,6 +91,9 @@
 
     // populate and show the modal
     $("#create-option-set .modal-body.option-set").load(self.params.new_option_set_path, function(){
+      var multilevel_checkbox = $("#option_set_multi_level");
+      self._can_be_multi_level() ? multilevel_checkbox.removeAttr('disabled') : multilevel_checkbox.attr('disabled', 'disabled')
```
Comment (hooverlunch) [inline app/assets/javascripts/views/question_form.js]: I see what you're doing here but I think that enabling/disabling the checkbox should be the responsibility of the OptionSetForm JS class. The question form really shouldn't know anything about that box, or whether an option set can be multilevel. Principle of least responsibility.

Instead I would pass a param to the OptionSetForm class indicating the question type for which the option set is getting created. Then let it decide how to handle itself.

You can do this by passing a GET param along with the .load call, which you can then pass to the OptionSetForm object on instantiation. (in option_sets/_form.html.erb).

Make sense?

**Comment by lmatiolis** [comment]
@hooverlunch Please check if this is what you meant. I did search for something to be set on the @option_set instance, but didn't find anything suitable to restrict the multi_level... is there something we can use? Right now, i'm just using the param directly on the partial... Thanks!

File: app/views/option_sets/_form.html.erb [hunk 2]
```diff
@@ -17,7 +17,7 @@
     <%= f.field(:geographic, :type => :check_box, :read_only => cannot?(:change_geographic, @option_set)) %>
     <%= f.field(:allow_coordinates, :type => :check_box, :read_only => cannot?(:change_geographic, @option_set)) %>
     <%= f.field(:multi_level, :type => :check_box,
-      :read_only => cannot?(:update_core, @option_set) || @option_set.has_select_multiple_questions? || @option_set.huge?) %>
+      :read_only => cannot?(:update_core, @option_set) || @option_set.has_select_multiple_questions? || @option_set.huge? || params[:question_type] == "select_multiple") %>
```
Comment (hooverlunch) [inline app/views/option_sets/_form.html.erb]: Yikes this is getting a bit hairy. How about this:
1. Add an attr_accessor called `adding_to_question_type` to OptionSet with a comment of 'Indicates that this OptionSet is being created to be added to a question of the given type'.
1. Change the GET param to match this.
1. In OptSetController, do `OptionSet.new(adding_to_question_type: params[:adding_to_question_type])` 
1. Add a new method to OptionSet called `can_be_multilevel?` that checks has_select_multiple_questions, huge, and adding_to_question_type.
1. Call that in the template.

Comment (lmatiolis) [inline app/views/option_sets/_form.html.erb]: Great, Tom! 
The same concern crossed my mind when i saw all those checks on the template... Doing what you said will improve this a lot!

Will do these changes here... Thanks!

**Pull request merged**